### PR TITLE
[editor] feat: use dropDownOptions prop from headingsPlugin to custom…

### PR DESCRIPTION
**Pull request by maimonav**
Many consumers have different demands when it comes to inline styles, headers in particular.
This PR uses `dropDownOptions` from `headingPlugin` to customize the headers types pasted from other editor inside ours.
If `headingPlugin` is not been used, then it checks for `headingMarkdownPlugin`, if it's not been used as well, it will disable all headers and all pasted headers will be converted into `unstyled` blocks.

### CODE REVIEW

File: packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
-  h1: 'header-one',
   h2: 'header-two',
   h3: 'header-three',
   h4: 'header-four',
   h5: 'header-five',
   h6: 'header-six',
 };
 
-const getBlockTypeOfElement = elementTag => headerElementToDraftType[elementTag] || 'unstyled';
+const headersArray = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js:11]:
```suggestion
const headersArray = ['h2', 'h3', 'h4', 'h5', 'h6'];
```

File: packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
+    // eslint-disable-next-line no-param-reassign
+    elementTag = 'h2';
+  }
+  const headerDraftType = headerElementToDraftType[elementTag];
+  if (customHeaders.includes(elementTag)) {
+    return headerDraftType;
+  }
+
+  if (headerDraftType) {
+    const headerIndex = headersArray.findIndex(header => header === elementTag);
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js:24]:
```suggestion
    const headerIndex = headersArray.indexOf(elementTag);
```

**Review by shaulgo — APPROVED** [review]

File: packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js (hunk 1 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js:11]: done.

File: packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js (hunk 2 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/utils/pasting/htmlToBlock.js:24]: done.

File: packages/editor/web/src/RichContentEditor/RichContentEditor.tsx [hunk 3]
```diff
@@ -529,12 +535,34 @@ class RichContentEditor extends Component<RichContentEditorProps, State> {
     }
   };
 
+  getHeadings = config => {
+    const {
+      [HEADINGS_DROPDOWN_TYPE]: headingsPluginSettings,
+      [HEADERS_MARKDOWN_TYPE]: headingsMarkdownSettings,
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/RichContentEditor.tsx:541]: check what markdown is. 
Also {$h1}sdafds{$h} is broken. And add test for it

**Review by shaulgo — APPROVED** [review]

**Pull request merged**